### ISSUE TITLE
Error when stop the Python WebSocket application with websocket-client version 1.2.1

### ISSUE THREAD

**Issue by wasin-waeosri** [thomsonreuters/websocket-api#16]
The Python examples (market_price.py, market_price_rdpgw_service_discovery.py, etc.) show the error message when pressing Ctrl+C to stop the application. This issue occurs with the websocket-client library version 1.2.1. The websocket-client library version 1.1.0 works fine without any error messages.

The market_price_rdpgw_service_discovery.py Example:
The websocket-client library version 1.2.1, running the application and pressing Ctrl+C after the application received an item refresh response message.
```
Closing the WebSocket connection for session1
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 407, in _callback
    callback(self, *args)
TypeError: _on_close() takes 2 positional arguments but 4 were given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 369, in run_forever
    dispatcher.read(self.sock.sock, read, check)
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 70, in read
    if not read_callback():
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 333, in read
    return teardown()
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 303, in teardown
    self._callback(self.on_close, close_status_code, close_reason)
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 412, in _callback
    self.on_error(self, e)
  File "market_price_rdpgw_service_discovery.py", line 175, in _on_error
    print(error + " for " + self.session_name)
TypeError: unsupported operand type(s) for +: 'TypeError' and 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 407, in _callback
    callback(self, *args)
  File "market_price_rdpgw_service_discovery.py", line 175, in _on_error
    print(error + " for " + self.session_name)
TypeError: unsupported operand type(s) for +: 'TypeError' and 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\threading.py", line 932, in _bootstrap_inner
    self.run()
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 371, in run_forever
    self._callback(self.on_error, e)
  File "C:\drive_d\compilers\Miniconda3\envs\testws\lib\site-packages\websocket\_app.py", line 412, in _callback
    self.on_error(self, e)
  File "market_price_rdpgw_service_discovery.py", line 175, in _on_error
    print(error + " for " + self.session_name)
TypeError: unsupported operand type(s) for +: 'TypeError' and 'str'
```
The websocket-client library version 1.1.0 does not show the error messages above.

The market_price.py Example:
The websocket-client library version 1.2.1 shows the following error message when closing the application. 
```
on_close() takes 1 positional argument but 3 were given
```
The websocket-client library version 1.1.0 does not show this error message above.

**Comment by vlevendel** [comment]
@wasin-waeosri This is a known issue with the following fix:

\- def _on_close(self, ws):
\+ def _on_close(self, ws, close_status_code, close_msg):

We will provide this change in a future release.

**Comment by vlevendel** [comment]
Fixed with tag WSA-1.1.8.
